File: Leetcode/2164. Sort Even and Odd Indices Independently/solution.go
import (
	"sort"
)

func sortEvenOdd(nums []int) []int {
	var res []int
	var odd []int
	var even []int

	for i, val := range nums {
		if i%2 == 0 {
			even = append(even, val)
		} else {
			odd = append(odd, val)
		}
	}
	sort.Sort(sort.IntSlice(even))
	sort.Sort(sort.Reverse(sort.IntSlice(odd)))

	for i, val := range odd {
		res = append(res, even[i])
		res = append(res, val)
	}

	if len(even) > len(odd) {
		res = append(res, even[len(even)-1])
	}

	return res
}

File: Leetcode/2164. Sort Even and Odd Indices Independently/solution.py
class Solution:
    def sortEvenOdd(self, nums: List[int]) -> List[int]:
        even_sorted = sorted([nums[n] for n in range(0, len(nums), 2)])
        odd_sorted = reversed(sorted([nums[n] for n in range(1, len(nums), 2)]))
        res = [0] * len(nums)
        i = 0
        for n in even_sorted:
            res[i] = n
            i += 2
        i = 1
        for n in odd_sorted:
            res[i] = n
            i += 2
        return res